src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -189,8 +192,43 @@ class BaseOptionsSelector extends Component {
         return defaultIndex;
     }
 
+    /**
+     * When pagination is enabled,
+     * maps sections to render only allowed count of them.
+     */
+    get sections() {
+        if (!_.isNumber(this.props.itemsPerPage)) {
+            return this.props.sections;
+        }
+
+        return _.map(this.props.sections, (section) => {
+            if (_.isEmpty(section.data)) {
+                return section;
+            }
+
+            return {
+                ...section,
+                data: section.data.slice(0, this.props.itemsPerPage * this.state.paginationPage),
+            };
+        });
+    }
+
+    /**
+     * Calculates all exactly visible options of sections.
+     */
+    get allVisibleOptionsCount() {
+        let count = 0;
+
+        _.forEach(this.sections, (section) => {
```

Comment:
did you mean `this.props.sections`? 

i can't find `sections` being defined anywhere in the class. Is it the `get` syntax binding it to the class?

```suggestion
        _.forEach(this.props.sections, (section) => {
```

Reply:
Yes, it is the `get` syntax. I will update it in order of removing the `get`.